contracts/ens/PodEnsRegistrar.sol

```diff
@@ -22,15 +26,27 @@ contract PodEnsRegistrar is Ownable {
      * @param ensAddr The address of the ENS registry.
      * @param node The node that this registrar administers.
      */
-    constructor(ENS ensAddr, Resolver resolverAddr, address _reverseRegistrar, IControllerRegistry controllerRegistryAddr, bytes32 node) {
+    constructor(ENS ensAddr, Resolver resolverAddr, address _reverseRegistrar, IControllerRegistry controllerRegistryAddr, bytes32 node, IInviteToken inviteTokenAddr) {
         ens = ensAddr;
         resolver = resolverAddr;
         controllerRegistry = controllerRegistryAddr;
         rootNode = node;
         reverseRegistrar = _reverseRegistrar;
+        inviteToken = inviteTokenAddr;
     }
 
-    function registerPod(bytes32 label, address podSafe) public returns(address) {
+    function registerPod(bytes32 label, address podSafe, address podCreator) public returns(address) {
+        // If podSafe and podCreator are the same value, then this is coming from the safe
+        // and therefore, it's a pod registration
+        if ((podSafe == podCreator) && restrictPodRegistration) {
```

Comment:
so the case that is missing is if `podSafe != podCreator`. A safe should be able to let one of its members register on behalf of the safe (this is better UX)

Reply:
maybe we just change the restrictions to `restrictToSafe` and `restrictToUser`
```js
    function registerPod(bytes32 label, address podSafe, address podCreator) public returns(address) {
       if (restrictToUser) {
            require(inviteToken.balanceOf(podCreator) > 0, "creator must have SHIP token");
            inviteToken.burn(podCreator, 1);
        } 
        if (restrictToSafe) {
            require(inviteToken.balanceOf(podSafe) > 0, "safe must have SHIP token");
            inviteToken.burn(podSafe, 1);
        }
```